File: 0448.找到所有数组中消失的数字/0448-找到所有数组中消失的数字.rs
impl Solution {
    pub fn find_disappeared_numbers(nums: Vec<i32>) -> Vec<i32> {
        let mut nums = nums;
        for i in 0..nums.len() {
            let index = (nums[i].abs() - 1) as usize;
            nums[index] = -(nums[index].abs());
        }
        let mut ret = vec![];
        for (i, &num) in nums.iter().enumerate() {
            if num > 0 {
                ret.push((i+1) as i32);
            }
        }
        
        ret
    }
}

File: 0448.找到所有数组中消失的数字/0448-找到所有数组中消失的数字.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        # return list(set((range(1,len(nums)+1))) - set(nums))
        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            nums[index] = -(abs(nums[index]))
        return [i + 1 for i, x in enumerate(nums) if x > 0]
